File: Problems/No.1877_Minimize_maximum_Pair_Sum_in_Array/minimize_maximum_pair_sum_in_array.py
class Solution(object):
    def minPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # parameter
        max_PS = 0
        i, j = 0, len(nums)-1

        # start computing
        nums.sort()
        while (i < j):
            max_PS = max(max_PS, nums[i]+nums[j])
            i += 1
            j -= 1
        
        return max_PS

File: Problems/No.1877_Minimize_maximum_Pair_Sum_in_Array/minimize_maximum_pair_sum_in_array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        
        // parameter
        int max_PS = 0;

        // start computing
        std::sort(nums.begin(), nums.end());
        for (int i=0, j=nums.size()-1; i<j; i++, j--) max_PS = std::max(max_PS, nums[i]+nums[j]);

        return max_PS;

    }
};
